pkg/kubelet/util/env/env_util_test.go

```diff
@@ -36,245 +68,483 @@ func TestParseEnv(t *testing.T) {
-KEY1=foo
-
-# third comment
-KEY2=bar
+			name: "newlines and tabs in value",
+			envContent: `KEY1='line1\nline2\tline3'
+KEY2='bar'
 `,
-			key:       "KEY2",
-			wantValue: "bar",
+			key:       "KEY1",
+			wantValue: `line1\nline2\tline3`,
 		},
 		{
-			name: "continuation with blank line in between (should error)",
-			envContent: `KEY1=foo \
-
-bar
+			name: "unicode characters in value",
+			envContent: `KEY1='中文 Español Français 🌍'
+KEY2='bar'
 `,
-			key:         "KEY1",
-			wantErr:     true,
-			errContains: "invalid environment variable format",
+			key:       "KEY1",
+			wantValue: `中文 Español Français 🌍`,
 		},
 		{
-			name: "continuation with comment in between (should error)",
-			envContent: `KEY1=foo \
-# comment
-bar
+			name: "backslashes in value",
+			envContent: `KEY1='path\\to\\file'
+KEY2='bar'
+`,
+			key:       "KEY1",
+			wantValue: `path\\to\\file`,
+		},
+		{
+			name: "quotes within quotes",
+			envContent: `KEY1='value with \'nested\' quotes'
+KEY2='bar'
 `,
 			key:         "KEY1",
 			wantErr:     true,
-			errContains: "invalid environment variable format",
+			errContains: "unexpected content after closing quote",
 		},
 		{
-			name: "empty value",
-			envContent: `KEY1=foo
-KEY2=
-KEY3=bar
+			name: "empty single quotes",
+			envContent: `KEY1=''
+KEY2=''
+KEY3='bar'
 `,
 			key:       "KEY2",
-			wantValue: "",
+			wantValue: ``,
 		},
 		{
-			name: "value with only spaces",
-			envContent: `KEY1=foo
-KEY2=   
-KEY3=bar
+			name: "only whitespace in quotes",
+			envContent: `KEY1='   '
+KEY2='\t\n'
+KEY3='bar'
 `,
-			key:       "KEY2",
-			wantValue: "",
+			key:       "KEY1",
+			wantValue: "   ",
 		},
 		{
-			name: "key is empty (should error)",
-			envContent: `=foo
-KEY2=bar
+			name: "complex JSON-like value",
+			envContent: `KEY1='{"name": "test", "value": 123, "nested": {"key": "val"}}'
+KEY2='bar'
 `,
-			key:         "KEY2",
-			wantValue:   "bar",
-			wantErr:     true,
-			errContains: "invalid environment variable format",
+			key:       "KEY1",
+			wantValue: `{"name": "test", "value": 123, "nested": {"key": "val"}}`,
 		},
 		{
-			name: "multiple = in value",
-			envContent: `KEY1=foo=bar=baz
-KEY2=bar
+			name: "URL with special characters",
+			envContent: `KEY1='https://example.com/path?query=value&another=param'
+KEY2='bar'
 `,
 			key:       "KEY1",
-			wantValue: "foo=bar=baz",
+			wantValue: `https://example.com/path?query=value&another=param`,
 		},
 		{
-			name: "continuation with trailing spaces",
-			envContent: `KEY1=foo   \
-  bar   \
-  baz   
-KEY2=bar
+			name: "XML-like content",
+			envContent: `KEY1='<root><element attr="value">content</element></root>'
+KEY2='bar'
 `,
 			key:       "KEY1",
-			wantValue: "foo bar baz",
+			wantValue: `<root><element attr="value">content</element></root>`,
 		},
 		{
-			name: "comment after key value pair",
-			envContent: `KEY1=foo
-KEY2=bar # comment
+			name: "base64 encoded data",
+			envContent: `KEY1='SGVsbG8gV29ybGQh'
```

Comment:
nit: Since it’s a Base64-encoded string, it might be better to use one that includes an `=` sign, such as the encoding of `Hello, world!`.

```suggestion
			envContent: `KEY1='SGVsbG8sIHdvcmxkIQ=='
```